pure_protobuf/protobuf.py

```diff
@@ -290,58 +342,65 @@ def __hash__(self):
         return _hash
 
     def __iter__(self):
-        '''
+        """
         Iterates over all fields.
-        '''
-        for tag, name in self.__tags_to_names.iteritems():
+        """
+        for tag, name in six.iteritems(self.__tags_to_names):
+            if not isinstance(name, str):
+                name = name.decode("utf-8")
```

Comment:
`name` is a field name, so byte string is not expected.

Reply:
See above with my comments on Py2 vs. Py3 here. Even setting names as byte literals produces strange behaviors.